packages/core-utils/src/watcher.ts

```diff
@@ -93,10 +98,15 @@ export class Watcher {
       const failureLogs = await layer.provider.getLogs(failureFilter)
       const logs = successLogs.concat(failureLogs)
       matches = logs.filter((log: ethers.providers.Log) => log.data === msgHash)
+
       // exit loop after first iteration if not polling
       if (!pollForPending) {
         break
       }
+
+      // pause awhile before trying again
+      const callback = (r) => setTimeout(r, this.pollInterval)
+      await new Promise(callback.bind(Watcher))
```

Comment:
```suggestion
      await new Promise(f => setTimeout(f, this.pollInterval));
```

Reply:
ah got the scope rules around `this` mixed up. thanks